File: 01.Array-and-TwoPointer/CPP-Solution/squareOfSortedArray.cpp
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        int n = nums.size();
        vector<int> ans(n);  // Initialize ans with the correct size
        int left = 0;
        int right = n - 1;

        for (int i = n - 1; i >= 0; i--) {
            if (nums[left] * nums[left] < nums[right] * nums[right]) {
                ans[i] = nums[right] * nums[right];
                right--;
            } else {
                ans[i] = nums[left] * nums[left];
                left++;
            }
        }
        return ans;
    }
};

File: 01.Array-and-TwoPointer/JavaScript-twoPointer-Solution/squareOfSortedArray.js
var sortedSquares = function(nums) {
    let n = nums.length;
    let ans__arr= Array(n);
    let left = 0;
    let right=n-1;
    for(let i=n-1 ;i>=0;i--){
        if(nums[left]**2<nums[right]**2){
            ans__arr[i]= nums[right]**2;
            right--;
        }else {
            ans__arr[i]=nums[left]**2;
            left++;
        }
    }
    return ans__arr;
};